Assignment 1:
Use get_balance_for_date to get balance for any date during the year starting from the value provided for the start of the year.

In [112]:
import pandas as pd
data = pd.read_csv('T1_input.csv')

In [113]:
data['date'] = pd.to_datetime(data['bookingDate']).dt.date

In [117]:
def get_balance_for_date(date):
    initial_amount_soy = 1773.80
    # use format 2018-12-31
    df = data.copy()
    df['earlier_than_date'] = data['date'] <= pd.to_datetime(date)
    rows = df.loc[df['earlier_than_date'] == True]
    amount = sum(rows['amount'])
    return sum([initial_amount_soy, amount])

In [187]:
balance_10_oct = get_balance_for_date('2018-10-01')
balance_31_jan = get_balance_for_date('2018-01-31')

In [189]:
balance_10_oct, balance_31_jan

(2600.6000000000004, 1787.9999999999998)

Use function as above to get the balance for a given date

Assignment dos:


In [148]:
data_two = pd.read_csv('T2_input.txt', delimiter="\t", header=None)

In [149]:
data_two['min'] = data_two.agg('min', axis='columns')
data_two['max'] = data_two.agg('max', axis='columns')
data_two['mean'] = data_two.agg('mean', axis="columns")

In [193]:
# get the mean of all means
mean_of_means = data_two['mean'].agg(['mean'], axis=0)
mean_of_means

mean    1478.635417
Name: mean, dtype: float64

In [194]:
# get which row has the lowest/highest mean
max_mean_row = data_two['mean'].argmax()
min_mean_row = data_two['mean'].argmin()
max_mean_row, min_mean_row

(11, 7)

In [195]:
data_two['mean'].describe()

count      16.000000
mean     1478.635417
std       802.677128
min       661.722222
25%       829.888889
50%      1085.916667
75%      2140.472222
max      2777.166667
Name: mean, dtype: float64

<strong>Results:</strong>

Mean of all means is 1478.64
<br>
Highest mean is on row 11
<br>
Lowest mean is on row 7